# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JSAtom.cpp
# Commit: 87ae5e3b6962
# Full Hash: 87ae5e3b6962b5db12bcb271a1fd2aef4e20a809
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2019-04-09 16:24:26
# Regressor Bug: 1540719
# File Overlap Count: 1
# Description:
#   Bug 1540719 - Perform a last ditch GC if symbol allocation fails as we do for most other GC things r=sfink
# ==============================================================================

diff -r 1b7309612d60 -r 87ae5e3b6962 js/src/vm/JSAtom.cpp
--- a/js/src/vm/JSAtom.cpp	Tue Apr 09 09:53:57 2019 +0300
+++ b/js/src/vm/JSAtom.cpp	Mon Apr 01 18:36:41 2019 +0100
@@ -293,8 +293,8 @@
   ImmutableSymbolPtr* symbols =
       reinterpret_cast<ImmutableSymbolPtr*>(wellKnownSymbols.ref());
   for (size_t i = 0; i < JS::WellKnownSymbolLimit; i++) {
-    JS::Symbol* symbol =
-        JS::Symbol::new_(cx, JS::SymbolCode(i), descriptions[i]);
+    HandlePropertyName description = descriptions[i];
+    JS::Symbol* symbol = JS::Symbol::new_(cx, JS::SymbolCode(i), description);
     if (!symbol) {
       ReportOutOfMemory(cx);
       return false;
@@ -930,8 +930,7 @@
   JSFlatString* flat = MakeFlatStringForAtomization(cx, chars, length);
   if (!flat) {
     // Grudgingly forgo last-ditch GC. The alternative would be to release
-    // the lock, manually GC here, and retry from the top. If you fix this,
-    // please also fix or comment the similar case in Symbol::new_.
+    // the lock, manually GC here, and retry from the top.
     ReportOutOfMemory(cx);
     return nullptr;
   }